### PR TITLE
Readme.md upgraded with process of starting vmaas up in developer mode

### PR DESCRIPTION
**Pull request by Michael-Jurek**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -133,23 +133,57 @@ docker-compose -f docker-compose.test.yml up --build --exit-code-from test
...
 ~~~
 
-switch inside of the container
+2.  Switch into **webapp** container:
+
+~~~bash
+docker-compose exec vmaas_webapp bash
+~~~
+
+> Now your local git directory is mounted under `/git` in the container so any change
```
Comment (jdobes) [inline README.md]: why is this text quoted?

File: README.md [hunk 2]
```diff
@@ -133,23 +133,57 @@ docker-compose -f docker-compose.test.yml up --build --exit-code-from test
...
+> you make you can immediatelly test.
+>
+> ```[root@localhost git]# python3 ./app.py```
+>
+> Note that by default container does NOT run the application
```
Comment (jdobes) [inline README.md]: same here, should not be quoted

File: README.md [hunk 3]
```diff
@@ -133,23 +133,57 @@ docker-compose -f docker-compose.test.yml up --build --exit-code-from test
...
+Now run:
 
-```[root@4cb6b50d0cb6 git]# python3 ./app.py```
 
-Note that by default container does NOT run the application (so you can run your own modificationtion)
-so if you want to run "original" (unmodified) application use
+```[root@localhost git]# ./entrypoint.sh```
 
-```[root@4cb6b50d0cb6 git]# /app/entrypoint.sh```
+> 5.    Switch to **database** container to gain access to db:
```
Comment (jdobes) [inline README.md]: same here, should not be quoted

**Comment by Michael-Jurek** [comment]
> Could you please mention that in reposcan you should run `rsync --daemon` before `./entrypoint.sh`?

entrypoint.sh already does `rsync --daemon` 
https://github.com/RedHatInsights/vmaas/blob/5b8584b5f9101f2921b864e3038cf2329605f9b7/reposcan/entrypoint.sh#L1-L6

File: README.md (hunk 2 above)
Comment (Michael-Jurek) [reply README.md]: OK

File: README.md (hunk 3 above)
Comment (Michael-Jurek) [reply README.md]: OK

File: README.md (hunk 1 above)
Comment (Michael-Jurek) [reply README.md]: Yeah, should not be commented.

File: README.md [hunk 4]
```diff
@@ -133,23 +133,59 @@ docker-compose -f docker-compose.test.yml up --build --exit-code-from test
 ~~~
 
 ### Developing / Debugging
-Build and start your container in "developer mode"
+
+1.  Build your container in "developer mode":
```
Comment (Josca) [inline README.md]: Now this is quite long guide with extended info. I would create extra page for that and link from `README.md`. This is not essential to run component.

File: README.md [hunk 5]
```diff
@@ -133,23 +133,59 @@ docker-compose -f docker-compose.test.yml up --build --exit-code-from test
...
+
+5.    Switch to **database** container to gain access to db:
+ 
+~~~bash
+docker-compose exec vmaas_database bash
+~~~
+
+Now you can run database terminal with:
+
+```[root@localhost git]# psql -d vmaas```
```
Comment (Josca) [inline README.md]: You have many extra spaces and empty lines here. It's not necessary. Could you remove it, please?

File: README.md [hunk 6]
```diff
@@ -133,23 +133,59 @@ docker-compose -f docker-compose.test.yml up --build --exit-code-from test
...
-./scripts/devel-compose exec vmaas_webapp bash
+docker-compose exec vmaas_webapp bash
 ~~~
 
-now your local git directory is mounted under `/git` in the container so any change
+Now your local git directory is mounted under `/git` in the container so any change
 you make you can immediatelly test.
 
-```[root@4cb6b50d0cb6 git]# python3 ./app.py```
+```[root@localhost git]# python3 ./app.py```
```
Comment (Josca) [inline README.md]: I would remove all `[root@localhost git]#` prompts and write only bash code. It complicates copy-paste process. You can also you bash notation and then use standard bash comments if you need:
~~~bash
python3 ./app.py # this is called in git repo
~~~

File: README.md [hunk 7]
```diff
@@ -133,23 +133,59 @@ docker-compose -f docker-compose.test.yml up --build --exit-code-from test
...
 so if you want to run "original" (unmodified) application use
 
-```[root@4cb6b50d0cb6 git]# /app/entrypoint.sh```
+```[root@localhost git]# /app/entrypoint.sh```
+
+Now you have bash inside vmaas_webapp container, run: 
+
+```[root@localhost git]# ./entrypoint.sh```
+
+3.  Switch into **websocket** container:
```
Comment (Josca) [inline README.md]: Unnecessary multiple spaces after `3.`.

File: README.md (hunk 4 above)
Comment (Michael-Jurek) [reply README.md]: OK, I'll do it.

File: README.md [hunk 8]
```diff
@@ -133,23 +133,4 @@ docker-compose -f docker-compose.test.yml up --build --exit-code-from test
...
-now your local git directory is mounted under `/git` in the container so any change
-you make you can immediatelly test.
-
-```[root@4cb6b50d0cb6 git]# python3 ./app.py```
-
-Note that by default container does NOT run the application (so you can run your own modificationtion)
-so if you want to run "original" (unmodified) application use
-
-```[root@4cb6b50d0cb6 git]# /app/entrypoint.sh```
+You can build and start your container in ["developer mode"](doc/build_containers.md).
```
Comment (Josca) [inline README.md]: Ok, but I would rename file to `developer_mode.md`. It's better regarding to its content. Also quotation marks are unnecessary.

File: doc/build_containers.md [hunk 9]
```diff
@@ -0,0 +1,56 @@
+# Develop with separate containers running
+1. Build your container in "developer mode":
+~~~bash
+./scripts/devel-compose build --no-cache
+./scripts/devel-compose up
+~~~
+
+2. Switch into **webapp** container:
+
```
Comment (Josca) [inline doc/build_containers.md]: Empty line to remove.

File: doc/build_containers.md [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+docker-compose exec vmaas_reposcan bash
+~~~
+
+Now run:
+
+
+```./entrypoint.sh```
+
+5. Switch to **database** container to gain access to db:
+
```
Comment (Josca) [inline doc/build_containers.md]: Empty line to remove.

File: doc/build_containers.md [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+```python3 websocket.py```
+
+4. Switch into **reposcan** container:
+~~~bash
+docker-compose exec vmaas_reposcan bash
+~~~
+
+Now run:
+
+
```
Comment (Josca) [inline doc/build_containers.md]: Empty line to remove.

**Pull request merged**